### DISCUSSION TITLE
libtensorflow_framework.so.2: cannot open shared object file: No such file or directory

### DISCUSSION THREAD

**Discussion by LeiLiu8102** [deepmodeling/deepmd-kit#248] [Q&A]
Hi, I have an error when installing the python interface of deepmd-kit.

I followed the instructions to install python interface of deepmd-kit and tensorflow:
`python3 -m venv --system-site-packages tf1.14`
`source tf1.14/bin/activate`
`pip install --upgrade pip`
`pip install --upgrade tensorflow`
`pip install --upgrade tensorflow==1.14`
`pip install deepmd-kit`

I verified my tensorflow installation by `python -c "import tensorflow as tf;print(tf.reduce_sum(tf.random.normal([1000, 1000])))"` and it outputs
>  Tensor("Sum:0", shape=(), dtype=float32)

When I execute the command `dp -h`,
it says
> tensorflow.python.framework.errors_impl.NotFoundError: libtensorflow_framework.so.2: cannot open shared object file: No such file or directory

Could you give any clue what the problem might be? Thank you.

I tried to address this issue by linking `libtensorflow_framework.so.2` to ` libtensorflow_framework.1`, because I can find ` libtensorflow_framework.so.1` in `tf1.14/lib64/python3.6/site-packages/tensorflow`. However it doesn't work.

System info:
python 3.6.8
gcc 4.8.5
cmake 3.11.0
tensorflow 1.14.0
deepmd 1.2.0

**Comment by njzjz** [comment]
Please upgrade Tensorflow to 2.1.

**Comment by LeiLiu8102** [comment]
Thank you for your comment. I upgrade Tensorflow to 2.1, then executed `dp -h`, and found the error
> Traceback (most recent call last):
  File "/home/whuan104/lliu147/tfvenv/bin/dp", line 5, in <module>
    from deepmd.main import main
  File "/home/whuan104/lliu147/tfvenv/lib64/python3.6/site-packages/deepmd/__init__.py", line 1, in <module>
    from .env import set_mkl
  File "/home/whuan104/lliu147/tfvenv/lib64/python3.6/site-packages/deepmd/env.py", line 61, in <module>
    op_module = get_module("libop_abi")
  File "/home/whuan104/lliu147/tfvenv/lib64/python3.6/site-packages/deepmd/env.py", line 58, in get_module
    module = tf.load_op_library(module_path + "{}.{}".format(module_name, ext))
  File "/home/whuan104/lliu147/tfvenv/lib64/python3.6/site-packages/tensorflow_core/python/framework/load_library.py", line 57, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename)
tensorflow.python.framework.errors_impl.NotFoundError: /home/whuan104/lliu147/tfvenv/lib/python3.6/site-packages/deepmd/libop_abi.so: undefined symbol: _ZN10tensorflow8OpKernel11TraceStringEPNS_15OpKernelContextEb

Could you please let me the possible reason? Thank you.

**Comment by njzjz** [comment]
I don't know why. [It was working in Google Colaboratory](https://nbviewer.jupyter.org/urls/drive.google.com/uc/%3Fexport%3Ddownload%26id%3D1afUT0ckcY57QfPskJdBESnIeHkljRPlS). You can try `pip install git+https://github.com/deepmodeling/deepmd-kit` to reinstall deepmd-kit.
